eng/Package.props

```diff
@@ -31,6 +31,9 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Update="Microsoft.Extensions.Logging" Version="$(MicrosoftExtensionsLoggingVersion)" />
```

Comment:
@MichaelSimons why this dependency was re-added? We are deliberately keeping it on lower version. If source build needs the latest, it should be added only for the source build.

Reply:
Can you help me understand why this is deliberately kept on a lower version?  Idealistically we would like to eliminate as many differences between source-build and the msft build.  To do this here we can either lift to the latest version which is how source-build works or make source-build use your pinned version.  A pinned older version requires source-build to define a ref package which comes at a cost and is something we would like to avoid.  This is described in https://github.com/dotnet/templating/issues/5177.

Reply:
Couple of reasons:
- allows to remove dependency on runtime to reduce PR churn
- we release all the assemblies as NuGet packages in addition to SDK, and I recall issues when the dep libs were not released leading to issues with packages
- our assemblies are also used in VS, and VS uses different version and it conflicts. Lower bound is easier to handle (if I'm not mistaken).

SDK anyway will use the latest version.

Reply:
@vlada-shubina Does that mean we're building against a version we aren't running against...this seems like a major risk.

Reply:
SDK uses latest version of  `Extensions.Logging`. Now we run tests also in SDK, and it's quite an exhaustive set, which also verifies logging.  Also, my understanding that normally there should be no breaking changes in `Extensions.Logging`, so the risk is not that high. 
Visual Studio uses different version of `Extensions.Logging` anyway.

Reply:
@vlada-shubina Does templating insert separately into VS? Is this about running on top of Framework in VS?

I worry about the tenability of this approach long term. Regardless of testing, it feels like bad practice. Looking forward to .NET 9, templating will be built in the VMR with the other repos (it cannot be built as a separate repo and flow into the SDK because it produces stable dependencies). I think that will mean that in the context of .NET Core, it must be built against the latest runtime packages.

What this might mean is that this repo will need to be more like msbuild, nuget, roslyn, etc. There will be an official build of those repos which is used to insert and ship to VS (and maybe NuGet.org?). Meanwhile, the source is inserted into the VMR building into .NET Core (SDK).

Reply:
Yes, the templating is inserted in VS and it's manual process. @phenning can tell more about this. VS use full framework version. 

There is a number of known external customers which use template engine using NuGet packages.